src/docs/reference/core/Configuration/README.md

```diff
@@ -54,6 +59,30 @@ In addition you can specify a `Tenant` setting by using the Tenant Name, in this
 }
 ```
 
+You can also configure `IOptions` from code in the web project's `Startup` class as explained in the [ASP.NET documentation](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options). 
+
+A lot of Orchard Core features are configured through the admin UI with site settings stored in the database. If you wish to override the site settings, you can do this with your own configuration code.
+
+For example, the Email module allows SMTP configuration via the `SmtpSettings` class which by default is populated from the given tenant's site settings, as set on the admin. However, you can override the site settings from the `Startup` class like this (note that we use `PostConfigure` to override the site settings' values but if the module doesn't use site settings you can just use `Configure`):
+
+```
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddOrchardCms(builder => builder.ConfigureServices(services =>
+        services.PostConfigure<SmtpSettings>(settings =>
+        {
+            // You could e.g. fetch the configuration values from and injected IShellConfiguration instance here.
+            settings.Port = 255;
+        })));
+}
```

Comment:
@Piedone @deanmarcussen 

Here you need another name for the local parameter `services` as it is already used by the enclosing scope, e.g.

    public void ConfigureServices(IServiceCollection services)
    {
        services.AddOrchardCms(builder => builder.ConfigureServices(tenantServices =>


Also now there is an `AddOrchardCms()` override that returns an `AddOrchardCms` so that you can use directly

    services.AddOrchardCms()
        .ConfigureServices(tenantServices =>

Reply:
@jtkech feel free to fix it directly from GH

Reply:
Okay will do, do we need to provide both ways? Or just fix this one

Reply:
Looking at it it's strange that it works with both variables being called `services` but this is actually valid code and there are no build errors. I've copied this from a VS window.

In the `Startup` class my personal preference is on this inner lambda but fine, I've changed it to the other one.

Reply:
> but this is actually valid code

On my side it doesn't allow it under VStudio, maybe i have other editor options but i use the ones defined in our repo

I would have used `tenantServices.PostConfigure()` or maybe `serviceCollection.PostConfigure()` or just `s.PostConfigure()` but not `configure.PostConfigure()`

Reply:
The beauty of having different approaches, isn't it? :) It's up to you if you'd like to change it.

Reply:
@Piedone no problem maybe this is my bad english that added some confusion ;)

I just wanted to say that there are 2 ways and pointed out the name issue

I also like the delegate one ;)

And in my last comment i was just talking about the parameter name you used here but anyway it's okay for me

    .ConfigureServices(configure => // here configure is a service collection

Reply:
No worries Jean-Thierry, thanks for pointing it out.